### ISSUE TITLE
bug/xlsx is wrong in API

### ISSUE THREAD

**Issue by leolorenzoluis** [Unstructured-IO/unstructured#2498]
**Describe the bug**
When using the public API, the XLSX output is wrong and it cuts out most of the text.

https://github.com/Unstructured-IO/unstructured/blob/79a1eb86837ccfd84eb265985d6981d9ef9fe965/unstructured/partition/xlsx.py#L62

Reproduced this in my local and it captures it properly

```
sheets = pd.read_excel(file_name, sheet_name=None)

for sheet_name, table in sheets.items():
    # print(f'table: {table}')
    html_text = table.to_html(index=False, header=False, na_rep="")
    # print(f'html_text: {html_text}')
    text = soupparser_fromstring(html_text).text_content()
    print(text)

```

So something is happening on the API level.

**To Reproduce**

```
file_name = 'Expenses.xlsx'
file_path = os.path.join(os.getcwd(), file_name)

with open(file_path, 'rb') as file:
   file_content = file.read()

  req = shared.PartitionParameters(
      # Note that this currently only supports a single file
      files=shared.Files(
          content=file_content,
          file_name=file_name,
      ),
      # Other partition params
      strategy="auto",
  )
```

**Expected behavior**

It should grab the entire text on the sheet

**Test file**
[test.xlsx](https://github.com/Unstructured-IO/unstructured/files/14149025/test.xlsx)

**Comment by Shashanks0616** [comment]
> **Describe the bug** When using the public API, the XLSX output is wrong and it cuts out most of the text.
> 
> https://github.com/Unstructured-IO/unstructured/blob/79a1eb86837ccfd84eb265985d6981d9ef9fe965/unstructured/partition/xlsx.py#L62
> 
> Reproduced this in my local and it captures it properly
> 
> ```
> sheets = pd.read_excel(file_name, sheet_name=None)
> 
> for sheet_name, table in sheets.items():
>     # print(f'table: {table}')
>     html_text = table.to_html(index=False, header=False, na_rep="")
>     # print(f'html_text: {html_text}')
>     text = soupparser_fromstring(html_text).text_content()
>     print(text)
> ```
> 
> So something is happening on the API level.
> 
> **To Reproduce**
> 
> ```
> file_name = 'Expenses.xlsx'
> file_path = os.path.join(os.getcwd(), file_name)
> 
> with open(file_path, 'rb') as file:
>    file_content = file.read()
> 
>   req = shared.PartitionParameters(
>       # Note that this currently only supports a single file
>       files=shared.Files(
>           content=file_content,
>           file_name=file_name,
>       ),
>       # Other partition params
>       strategy="auto",
>   )
> ```
> 
> **Expected behavior** A clear and concise description of what you expected to happen.
> 
> It should grab the entire text on the sheet
> 
> **Test file** [test.xlsx](https://github.com/Unstructured-IO/unstructured/files/14149025/test.xlsx)
Instead of converting the DataFrame to HTML and then parsing it back to text, we directly convert the DataFrame to text using the to_string() method. This ensures that the text is not truncated or cut off during the conversion process.
We also removed the unnecessary conversion to HTML in the metadata creation when include_metadata is True, as the HTML text is not being used.
Additionally, I've made a minor adjustment to the file parameter type annotation to accept BinaryIO instead of SpooledTemporaryFile, as it seems more appropriate for the provided use case.

**Comment by scanny** [comment]
@leolorenzoluis and @Shashanks0616 I believe this is the same bug as here: https://github.com/Unstructured-IO/unstructured/issues/2485. Can you have a look and see whether that possibly describes your case?

The fix for that is code-complete and will be in a PR hopefully by the end of the day. I'll let you know so you can try installing from that branch and see if that fixes your problem.

**Comment by scanny** [comment]
Fixed by #2534.
